docs/timeline.md

```diff
@@ -20,14 +20,26 @@ performance unless your application is run in profile mode.
 
 This chart is populated with individual frames as they are rendered in your application. Each bar in the chart
 represents a frame. The bars are color-coded to highlight the different portions of work that occur when rendering a
-Flutter frame:
-- CPU: Dart code running from your application as well as the the Flutter framework
-- GPU: graphics code running from the Flutter Engine
+Flutter frame: work from the CPU thread and work from the GPU thread.
 
 <img src="images/timeline_frame_rendering_chart.png" width="800" />
 
 Clicking a bar will display additional details about the frame.
 
+### CPU
+
+The CPU thread (or UI thread) executes Dart code in the Dart VM. This includes code from your application as well as the
+Flutter framework. When your app creates and displays a scene, the CPU thread creates a layer tree, a lightweight object
+containing device-agnostic painting commands, and sends the layer tree to the GPU thread to be rendered on the device.
+Do **not** block this thread.
+
+### GPU
+
+The GPU thread executes graphics code from the Flutter Engine. This thread takes the layer tree and displays it by
+talking to the GPU (graphic processing unit). You cannot directly access the GPU thread or its data, but if this thread
+is slow, it’s a result of something you’ve done in the Dart code. Skia, the graphics library, runs on this thread, which
```

Comment:
Can you give some examples of GPU thread issues?
A simple example could be if you go overboard and create a ton of RepaintBoundaries you will cause the GPU to do a lot more work. The flutter team probably has some better cannonical cases of ways you make the GPU do too much work.

Reply:
Added info from https://flutter.dev/docs/testing/ui-performance#identifying-problems-in-the-gpu-graph and liked out to flutter.dev for more info.